### PR TITLE
Main

### PR DESCRIPTION
**Pull request by BonedDeng**
Add explicit random seed support to ensure reproducibility of results across runs. This helps avoid discrepancies caused by non-deterministic behavior during data loading, splitting, or model initialization.

add set_seed on chemprop/utils/utils.py
check on training.ipynb

### CODE REVIEW

**Comment by JacksonBurns** [comment]
We appreciate the PR @BonedDeng. `lightning` already has a function `seed_everything` that takes care of `numpy`, `torch`, and `lightning` seeds [here](https://github.com/Lightning-AI/pytorch-lightning/blob/ab7b29950fcb090ec3712f1fe788a0ba8605bb7f/src/lightning/fabric/utilities/seed.py#L20) and the `Trainer` class has a deterministic option that takes care of the rest [here](https://github.com/Lightning-AI/pytorch-lightning/blob/master/src/lightning/pytorch/trainer/connectors/accelerator_connector.py#L626). All of that functionality is already exposed through the CLI. We'll let users decide on using the functionality already available when using `chemprop` from code rather than providing another setup ourselves.

**Comment by KnathanM** [comment]
@BonedDeng if you find that using the options Jackson described doesn't give you deterministic results, please open an issue so we can try to address that.
